File: xcod2json.py
import json
jsondata = {
"data": [],
"textures": []
}
filename = input("Enter xcod name: ")
with open(filename, "rb") as fh:
	data = fh.read().hex()
	picCount = int(data[10:12], 16)
	parse = 7
	parseWidth = 8
	parseWidth2 = 9
	parseHeight = 10
	parseHeight2 = 11
	for i in range(picCount):
		fh.seek(parse)
		pixelType = fh.read(1)
		fh.seek(parseWidth)
		width1 =  fh.read(1)
		width1 = int.from_bytes(width1, byteorder="little")
		width1 *= 256
		fh.seek(parseWidth2)
		width2 =  fh.read(1)
		width2 = int.from_bytes(width2, byteorder="little")
		fh.seek(parseHeight)
		height1 =  fh.read(1)
		height1 = int.from_bytes(height1, byteorder="little")
		height1 *= 256
		fh.seek(parseHeight2)
		height2 =  fh.read(1)
		height2 = int.from_bytes(height2, byteorder="little")
		height = height1 + height2
		width = width1 + width2
		pixelType = int.from_bytes(pixelType, byteorder="little")
		texture = {
		"pixelType": pixelType,
		"width": width,
		"height": height
		}
		jsondata["textures"].append(texture)
		parse += 6
		parseHeight += 6
		parseHeight2 += 6
		parseWidth += 6
		parseWidth2 += 6
picData = {
"pictureCount": picCount
}
jsondata["data"].append(picData)
jsonfile = json.dumps(jsondata, indent=4)
filename = filename.replace(".xcod", ".json")
with open(filename, "w") as fh:
	fh.write(jsonfile)

File: xcod2json.js
const fs = require('fs');
let jsondata = {
  data: [],
  textures: []
};

const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter xcod name: ', (filename) => {
  fs.readFile(filename, 'hex', (err, data) => {
    if (err) {
      console.error(err);
      rl.close();
      return;
    }
    const picCount = parseInt(data.slice(10, 12), 16);
    let parse = 7;
    let parseWidth = 8;
    let parseWidth2 = 9;
    let parseHeight = 10;
    let parseHeight2 = 11;

    for (let i = 0; i < picCount; i++) {
      const pixelType = parseInt(data.slice(parse * 2, (parse * 2) + 2), 16);
      const width1 = parseInt(data.slice(parseWidth * 2, (parseWidth * 2) + 2), 16);
      const width2 = parseInt(data.slice(parseWidth2 * 2, (parseWidth2 * 2) + 2), 16);
      const height1 = parseInt(data.slice(parseHeight * 2, (parseHeight * 2) + 2), 16);
      const height2 = parseInt(data.slice(parseHeight2 * 2, (parseHeight2 * 2) + 2), 16);
      const width = (width1 * 256) + width2;
      const height = (height1 * 256) + height2;

      const texture = {
        pixelType: pixelType,
        width: width,
        height: height
      };
      jsondata.textures.push(texture);

      parse += 6;
      parseHeight += 6;
      parseHeight2 += 6;
      parseWidth += 6;
      parseWidth2 += 6;
    }

    const picData = {
      pictureCount: picCount
    };
    jsondata.data.push(picData);
    const jsonfile = JSON.stringify(jsondata, null, 4);
    const newFilename = filename.replace('.xcod', '.json');
    fs.writeFile(newFilename, jsonfile, (err) => {
      if (err) {
        console.error(err);
      } else {
        console.log('JSON file has been written successfully.');
      }
      rl.close();
    });
  });
});
